spring-cloud-gcp-samples/spring-cloud-gcp-pubsub-binder-sample/src/main/java/com/example/PubSubBinderApplication.java

```diff
@@ -18,13 +18,20 @@
 
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.cloud.gcp.pubsub.support.converter.SimplePubSubMessageConverter;
+import org.springframework.context.annotation.Bean;
 
 /**
  * @author João André Martins
  */
 @SpringBootApplication
 public class PubSubBinderApplication {
 
+	@Bean
+	public SimplePubSubMessageConverter messageConverter() {
```

Comment:
Wait. What is wrong with existing configuration?
The current `PubSubInboundChannelAdapter` has this by default:
```
private PubSubMessageConverter pubSubMessageConverter = new SimplePubSubMessageConverter();
```

So, what am I missing?

However I can guess that an issue is fully related to what @meltsufin has just reported with the #865.

On the other side I fully agree that it's not OK to rely on some pre-configured converter. According Spring Cloud Stream logic we always have to produce `byte[]` from the Binder: https://docs.spring.io/spring-cloud-stream/docs/Elmhurst.RELEASE/reference/htmlsingle/#content-type-management

I remember as I mentioned that once, but it's hard to dig a conversion log now.
So, my point is that this is really a good catch, but it must be fixed explicitly with some `PathThroughConverter` to return just plain `byte[]` or change a `PubSubInboundChannelAdapter` back to the state where we can configure a converter to `null` to just return `byte[]` as is without any conversion attempts.